libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -795,11 +795,17 @@ export class CartService {
     ]);
     const cartEligibilityStatus =
       handleEligibilityStatusMap[eligibility.subscriptionEligibilityResult];
+    const { unitAmountForCurrency: offeringPrice } =
+      await this.priceManager.retrievePricingForCurrency(
+        price.id,
+        cart.currency
+      );
+    assertNotNull(offeringPrice);
```

Comment:
Switch this over to an explicit error, if you don't mind. Its easier to debug than AssertionError's are